src/cantypes.hpp

```diff
@@ -5,23 +5,35 @@
 #include <map>
 #include <string>
 #include <vector>
+#include <math.h>
+#include <boost/optional.hpp>
+#include <boost/any.hpp>
 
-enum class CANsignalType { Int, Float, String };
+enum class CANsignalType { Unknown = -1, SignedUnsignedInt = 0, Float = 1,
+    Double = 2, Count };
+enum class CANsignalMuxType { NotMuxed = 0, Muxer, Muxed };
+enum class CANsignalEndianness { BigEndianMotorola = 0, LittleEndianIntel = 1 };
 
 struct CANsignal {
     std::string signal_name;
     std::uint8_t startBit;
     std::uint8_t signalSize;
-    std::uint8_t byteOrder;
+    CANsignalEndianness endianness;
     bool valueSigned;
-    float factor;
-    float offset;
-    float min;
-    float max;
+    std::double_t factor;
+    std::double_t offset;
+    std::double_t min;
+    std::double_t max;
     std::string unit;
-    std::vector<std::string> receiver;
-    std::string mux = "";
-    std::uint8_t muxNdx{ 0 };
+    std::vector<std::string> receivers;
+    CANsignalMuxType muxType{ CANsignalMuxType::NotMuxed };
+    // 16-bit for better debug printing via boost::optional
+    boost::optional<std::uint16_t> muxNdx{ boost::none };
+    boost::optional<boost::any> startValue { boost::none };
```

Comment:
should this be boost:optional<double> ?

Reply:
Yeah, because the start value is optional. It is not part of a signal definition and only optionally appears later in the file, in a form like this:

```
BA_ "GenSigStartValue" SG_ 100 DRIVER_HEARTBEAT_cmd 3;
```